MoreLinq.Test/InterleaveTest.cs

```diff
@@ -24,35 +24,41 @@ public void TestInterleaveIsLazy()
         /// Verify that invoking Interleave on a <c>null</c> sequence results in an exception
         /// </summary>
         [Test]
-        [ExpectedException(typeof(ArgumentNullException))]
         public void TestInterleaveNullSequenceArgument()
         {
-            const IEnumerable<int> sequence = null;
-            sequence.Interleave(new int[] { });
+            Assert.Throws<ArgumentNullException>(() =>
+            {
+                const IEnumerable<int> sequence = null;
+                sequence.Interleave(new int[] { });
+            });
         }
 
         /// <summary>
         /// Verify that invoking Interleave with a <c>null</c> otherSequences parameter results in an exception
         /// </summary>
         [Test]
-        [ExpectedException(typeof(ArgumentNullException))]
         public void TestInterleaveNullOtherSequencesArgument()
         {
-            const int count = 10;
-            var sequence = Enumerable.Range(1, count);
-            sequence.Interleave(null);
+            Assert.Throws<ArgumentNullException>(() =>
+            {
+                const int count = 10;
+                var sequence = Enumerable.Range(1, count);
+                sequence.Interleave(null);
```

Comment:
Only this line should be inside the lambda to `Assert.Throws` as that's what will cause and being tested for an exception.